BuildConfigGen/Program.cs

```diff
@@ -375,6 +376,20 @@ private static void WriteNodePackageJson(string taskOutputNode, string nodeVersi
             ensureUpdateModeVerifier!.WriteAllText(outputNodePackagePath, nodePackageContent, suppressValidationErrorIfTargetPathDoesntExist: false);
         }
 
+        private static void WriteDevDependencyPackageJson(string taskOutputNode, string nodeVersion)
+        {
+            string outputNodePackagePath = Path.Combine(taskOutputNode, "package.json");
+            JsonNode outputNodePackagePathJsonNode = JsonNode.Parse(ensureUpdateModeVerifier!.FileReadAllText(outputNodePackagePath))!;
+            
+            // Upgrade typescript version for Node 20
+            if (nodeVersion == "^20.3.1") {
+                outputNodePackagePathJsonNode["devDependencies"]!["typescript"] = "4.8.4";
+            }
+            string nodePackageContent = outputNodePackagePathJsonNode.ToJsonString(jso) + Environment.NewLine;
```

Comment:
Please carry over the comments above regarding this new line, otherwise it's not obvious why this new line is present:
            // We need to add newline since npm install command always add newline at the end of package.json
            // https://github.com/npm/npm/issues/18545